### ISSUE TITLE
Optional properties are assigned inverse?

### ISSUE THREAD

**Issue by florianmartens** [Eywek/oatyp#7]
This is such a great library that saves me a ton of time. I'm super grateful it exists! I have a small issue as illustrated below:

**profile.entitiy.ts**
```
export class ProfileEntity implements Profile {
    @ApiProperty()
    id: string;

    @ApiProperty()
    createdAt: Date;

    @ApiProperty()
    updatedAt: Date;

    @ApiProperty({ required: false })
    address: string;

    @ApiProperty({ required: false })
    city: string;

    @ApiProperty({ required: false })
    countryCode: string;

    @ApiProperty({ required: false })
    phone: string;

    @ApiProperty({ required: false })
    twitterLink: string;

    @ApiProperty({ required: false })
    linkedInLink: string;

    @ApiProperty()
    userId: string;
}
```

Generated type
```
export type ProfileEntity = {
    id?: string;
    createdAt?: Date;
    updatedAt?: Date;
    address: string;
    city: string;
    countryCode: string;
    phone: string;
    twitterLink: string;
    linkedInLink: string;
    userId?: string;
};
```

It seems to be as if the optional `?` is appended inverse to the way it should be {required: false} should result in the type being optional and vice versa.

This happens to all my types (the above is just an example).

**Comment by vmarchaud** [comment]
Which library are you using the translate the decorators in openapi definition ? Could you share the definition to be able to reproduce the issue ?

**Comment by florianmartens** [comment]
Of course. I'm using NestJS which includes the package `@nestjs/swagger`.

I created a gist with my swagger.json file [here](https://gist.github.com/florianmartens/e0a264062fea931b61c4a5640d53ef45).

**Comment by vmarchaud** [comment]
Indeed this should be reversed from the openapi definition. Do you know if you have time to investigate this ?

**Comment by florianmartens** [comment]
I think that it would probably take me some time to figure out what's going in this project. If it's just a super quick fix like a missing inversion, the library creator or maintainer might be kind enough to quickly fix it.

Alternatively, I might find some time in the future to poke around in the code but can't make that commitment as of now.

**Comment by vmarchaud** [comment]
I was asking because i know that he's in holiday right now so if that was blocking/urgent, that may have been faster to make a PR direclty but no worries

**Comment by Eywek** [comment]
Hello,

It should be fixed in the `1.0.0-alpha.20` version, can you confirm?

**Comment by florianmartens** [comment]
Hey @Eywek , I can confirm that the fix works. Thank you!
